addons/website/static/src/js/tours/tour_utils.js

```diff
@@ -400,6 +400,31 @@ function selectElementInWeSelectWidget(widgetName, elementName, searchNeeded = f
     return steps;
 }
 
+/**
+ * Switches to a different website by clicking on the website switcher.
+ *
+ * @param {number} websiteId - The ID of the website to switch to.
+ * @param {string} websiteName - The name of the website to switch to.
+ * @returns {Array} - The steps required to perform the website switch.
+ */
+function switchWebsite(websiteId, websiteName) {
+    return [{
+        content: `Click on the website switch to switch to website '${websiteName}'`,
+        trigger: '.o_website_switcher_container button',
+    }, {
+        content: `Switch to website '${websiteName}'`,
+        extra_trigger: `iframe html:not([data-website-id="${websiteId}"])`,
+        trigger: `.o_website_switcher_container .dropdown-item:contains("${websiteName}")`,
+    }, {
+        content: "Wait for the iframe to be loaded",
+        // The page reload generates assets for the new website, it may take
+        // some time
+        timeout: 20000,
+        trigger: `iframe html[data-website-id="${websiteId}"]`,
+        isCheck: true,
+    }];
+}
```

Comment:
Why clicking on the website switcher **then** checking you are not already on the website you are targeting? Also is this useful to actually check you are not already on the targeted website? I guess it's ok but feels like the util would be more flexible without it.

Reply:
> I guess it's ok but feels like the util would be more flexible without it.

Would there be any legit case to click on the website you already are on? Sounds like an overkill protection indeed but since it was written like that in the first place and since this tour apparently never caused any race condition or issue (other than the timeout), I just kept the diff as close as possible as what it was.

> why clicking on the website switcher then checking you are not already on the website you are targeting? 

I hesitated to change that, felt weird to me too. Kept the diff to the minimum, I can change it if you want

Reply:
Ok, sure :+1: Ping me once retargeted to 16.0 :+1: 

> Would there be any legit case to click on the website you already are on?

Said like that, no :wink: But could be used as an "ensureWebsite(x)" function. Kinda like the "click on extra menu if it exists" util. But we should probably avoid that now that I think about it.